app/models/scimitar/resources/mixin.rb

```diff
@@ -417,7 +408,7 @@ def from_scim!(scim_hash:)
         #
         def from_scim_patch!(patch_hash:)
           frozen_ci_patch_hash = patch_hash.with_indifferent_case_insensitive_access().freeze()
-          ci_scim_hash         = self.to_scim(location: '(unused)').as_json().with_indifferent_case_insensitive_access()
+          ci_scim_hash         = self.to_scim_for_patch_op().as_json().with_indifferent_case_insensitive_access()
```

Comment:
Idiomatic SCIM resource controller code for an `#update` operation has this at the core:

```ruby
record = self.find_record(record_id)
record.from_scim_patch!(patch_hash: patch_hash)
self.save!(record, &block)
# ...and render 'record' in SCIM form...
```

That's how we get to the method being changed here. At the end of this method - you'd need to expand the diff downwards to around new diff line 485 or so - the final steps are:

```ruby
self.from_scim!(scim_hash: ci_scim_hash)
return self
```

The idea is that we build `ci_scim_hash` from the existing data as per the line above, then modify it via whatever update operations are requested, then repopulate the resource from that updated hash. This is related to long-standing issue #6. In lieu of the workload required to produce a true fix, and given that the mixin's tests do include tests to say "if a value is omitted in the Hash then it _is_ set to `nil`", we can't allow `ci_scim_hash` to have missing keys due to non-rendered "returned never" attributes. Instead, it _must_ contain them all, so that the current relatively crude (but still, thanks to SCIM's design, very complex) implementation can work as expected.

I note in passing that in the special case of a password, when something like Devise is in use and being used properly (!) then, once a password is written it is encrypted and cannot be read. Attempts to read the password or confirmation return nil; attempts to _write_ nil or empty string have no side effects and are ignored. This is the _desired outcome_ for that attribute. It does open up the question of non-password "returned never" attributes where the underling model might not allow a read operation, though, causing set-to-nil behaviour in subsequent SCIM patch updates.

TL;DR, what a mess. But this gets us by for now, and resolves issues highlighted by the updated tests.